### PR TITLE
Acceptance tests: increase api-gateway retries

### PR DESCRIPTION
**Pull request by curtbushko**
Changes proposed in this PR:
- Increase retry timing and add a retry to getting config entries
- I still saw one flake related to xDS rate limiting but I wasn't able to reproduce it

How I've tested this PR:

- ran it through CI several times

How I expect reviewers to test this PR:

:eyes:

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 1]
```diff
@@ -115,11 +120,6 @@ func TestAPIGateway_Basic(t *testing.T) {
 				k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "-f", "../fixtures/cases/api-gateways/tcproute/route.yaml")
 			})
 
-			// We use the static-client pod so that we can make calls to the api gateway
-			// via kubectl exec without needing a route into the cluster from the test machine.
-			logger.Log(t, "creating static-client pod")
-			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/bases/static-client")
-
```
Comment (zalimeni) [inline acceptance/tests/api-gateway/api_gateway_test.go:122]: ~ Why the relocation above for this block?

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 2]
```diff
@@ -209,17 +209,25 @@ func TestAPIGateway_Basic(t *testing.T) {
 			checkStatusCondition(t, tcpRoute.Status.Parents[0].Conditions, trueCondition("ConsulAccepted", "Accepted"))
 
 			// check that the Consul entries were created
-			entry, _, err := consulClient.ConfigEntries().Get(api.APIGateway, "gateway", nil)
-			require.NoError(t, err)
-			gateway := entry.(*api.APIGatewayConfigEntry)
+			// On startup, the controller can take upwards of 1m to perform
+			// leader election so we may need to wait a long time for
+			// the reconcile loop to run (hence the 1m timeout here).
```
Comment (zalimeni) [inline acceptance/tests/api-gateway/api_gateway_test.go:214]: It looks like we have a retry above w/ the same comment - do we expect this second block to fail independently after the first passes for the same leader election, or am I missing some nuance?

**Review by zalimeni — APPROVED** [review]
Few non-blocking questions, LMK if any of them don't make sense. Otherwise LGTM.

It seems like at worst we'll pick up the difference for a single retry check here in terms of failing more slowly when things flake, so if it buys us fewer timeout-related flakes, that's a win.

File: acceptance/tests/api-gateway/api_gateway_test.go (hunk 1 above)
Comment (curtbushko) [reply acceptance/tests/api-gateway/api_gateway_test.go:122]: I was shuffling things around to add retries & checks to the deployments but it didn't work out.

File: acceptance/tests/api-gateway/api_gateway_test.go (hunk 2 above)
Comment (curtbushko) [reply acceptance/tests/api-gateway/api_gateway_test.go]: Removed the comment.

File: acceptance/tests/api-gateway/api_gateway_test.go (hunk 1 above)
Comment (zalimeni) [reply acceptance/tests/api-gateway/api_gateway_test.go:122]: Oh, like we don't need to move this after all?

Comment (curtbushko) [reply acceptance/tests/api-gateway/api_gateway_test.go:122]: Not really. It'll give it a few ms to deploy and get ready but it shouldn't make a difference.

**Pull request merged**